infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/LogReplicationSourceManager.java

```diff
@@ -61,119 +70,89 @@
-     *                   the application callback for data transmission
-     * @param params Log Replication Parameters
-     * @param logReplicationFSMWorkers worker thread pool (state tasks)
-     */
-    public LogReplicationSourceManager(CorfuRuntime runtime,
-                                       DataSender dataSender,
-                                       LogReplicationRuntimeParameters params,
-                                       ExecutorService logReplicationFSMWorkers) {
-        this(runtime, dataSender, new DefaultReadProcessor(runtime), params, logReplicationFSMWorkers);
-    }
-
-    /**
-     * Constructor Source to provide ExecutorServices for FSM
-     *
-     * For multi-cluster log replication multiple managers can share a common thread pool.
-     *
-     * @param runtime corfu runtime
-     * @param dataSender implementation of a data sender, both snapshot and log entry, this represents
-     *                   the application callback for data transmission
-     * @param readProcessor implementation for reads processor (transformation)
-     * @param params Log Replication Parameters
-     * @param logReplicationFSMWorkers worker thread pool (state tasks)
-     */
-    public LogReplicationSourceManager(CorfuRuntime runtime,
-                                       DataSender dataSender,
-                                       ReadProcessor readProcessor,
-                                       LogReplicationRuntimeParameters params,
-                                       ExecutorService logReplicationFSMWorkers) {
+                .tlsEnabled(params.isTlsEnabled()).build());
+        runtime.parseConfigurationString(params.getLocalCorfuEndpoint()).connect();
 
         this.parameters = params;
+
         this.config = parameters.getReplicationConfig();
         if (config.getStreamsToReplicate() == null || config.getStreamsToReplicate().isEmpty()) {
             // Avoid FSM being initialized if there are no streams to replicate
             throw new IllegalArgumentException("Invalid Log Replication: Streams to replicate is EMPTY");
         }
 
+        ExecutorService logReplicationFSMWorkers = Executors.newFixedThreadPool(DEFAULT_FSM_WORKER_THREADS, new
+                ThreadFactoryBuilder().setNameFormat("state-machine-worker").build());
+        ReadProcessor readProcessor = new DefaultReadProcessor(runtime);
+
         // If this runtime has opened other streams, it appends non opaque entries and because
         // the cache is shared we end up doing deserialization. We need guarantees that this runtime is dedicated
         // for log replication exclusively.
-        this.runtime = CorfuRuntime.fromParameters(runtime.getParameters());
-        this.runtime.parseConfigurationString(runtime.getLayoutServers().get(0)).connect();
+        //this.runtime = CorfuRuntime.fromParameters(runtime.getParameters());
+        //this.runtime.parseConfigurationString(runtime.getLayoutServers().get(0)).connect();
```

Comment:
Should we remove these lines? as well as the above comments?

Reply:
+1